### PR TITLE
[Refactor] Support passing arguments to loss from head.

### PR DESCRIPTION
**Pull request by Ezra-Yu**
## Motivation

the forward function of loss(CE, Focal Loss and etc) have parameters like `weight` and so on. but users can not pass `weight` by using function head.forward_train.

## Modification

Add **kwargs in heads, and add kwargs in loss function in Clshead.

## BC-breaking (Optional)

No.

## Use cases (Optional)

```
weight = torch.tensor( [0.5, 0.5] )
head.forward_train(imgs, gt_labels, weight=weight........)

```

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by mzr1996 — APPROVED** [review]
LGTM
